### PR TITLE
Screen Narration stuff

### PR DESCRIPTION
**Pull request by DreenDex**
I have _concerns_ about `ToStringTransformer`'s package and `Narratable` name. Here are some options:

`Narratable` can also be named either `Narrative` or `Announceable`.  
Also I put new `ToStringTransformer` class in `net/minecraft/util`, but another suitable package is `net/minecraft/text`.

### CODE REVIEW

File: mappings/net/minecraft/util/ToStringTransformer.mapping [hunk 1]
```diff
@@ -0,0 +1,23 @@
+CLASS net/minecraft/class_6384 net/minecraft/util/ToStringTransformer
```
Comment (liach) [inline mappings/net/minecraft/util/ToStringTransformer.mapping]: Imo this should go to `gui/screen/narration` as this is used to generate sentences for any arbitrary input. When I mapped this, I called this `Narration` for this contains the data for narration in `value`; and I named the `transformer` a `feeder` as it feeds sentences into the given consumer (seen as a reader)

File: mappings/net/minecraft/client/gui/screen/narration/NarrationMessageBuilder.mapping [hunk 2]
```diff
@@ -0,0 +1,14 @@
+CLASS net/minecraft/class_6382 net/minecraft/client/gui/screen/narration/NarrationMessageBuilder
+	METHOD method_37031 nextMessage ()Lnet/minecraft/class_6382;
+	METHOD method_37032 putPart (Lnet/minecraft/class_6381;Lnet/minecraft/class_6384;)V
```
Comment (liach) [inline mappings/net/minecraft/client/gui/screen/narration/NarrationMessageBuilder.mapping]:
```suggestion
	METHOD method_37032 put (Lnet/minecraft/class_6381;Lnet/minecraft/class_6384;)V
```

File: mappings/net/minecraft/client/gui/screen/Screen.mapping [hunk 3]
```diff
@@ -10,6 +10,17 @@ CLASS net/minecraft/class_437 net/minecraft/client/gui/screen/Screen
...
 	FIELD field_22793 textRenderer Lnet/minecraft/class_327;
+	FIELD field_33813 selected Lnet/minecraft/class_6379;
+	FIELD field_33814 NARRATOR_SCREEN_USAGE Lnet/minecraft/class_2561;
+	FIELD field_33815 selectables Ljava/util/List;
+	FIELD field_33816 drawables Ljava/util/List;
+	FIELD field_33817 SCREEN_INIT_NARRATION_DELAY J
+	FIELD field_33818 NARRATOR_MODE_CHANGE_DELAY J
+	FIELD field_33822 narrator Lnet/minecraft/class_6385;
+	FIELD field_33823 elementNarrationStartTimeMs J
+	FIELD field_33824 screenNarrationStartTimeMs J
```
Comment (liach) [inline mappings/net/minecraft/client/gui/screen/Screen.mapping]:
```suggestion
	FIELD field_33823 elementNarrationStartTime J
	FIELD field_33824 screenNarrationStartTime J
```
We usually assume time are in milliseconds

File: mappings/net/minecraft/client/gui/screen/Screen.mapping [hunk 4]
```diff
@@ -96,7 +103,7 @@ CLASS net/minecraft/class_437 net/minecraft/client/gui/screen/Screen
 		COMMENT Renders the fullscreen {@linkplain net.minecraft.client.gui.DrawableHelper#OPTIONS_BACKGROUND_TEXTURE background texture} of this screen.
 		ARG 1 vOffset
 			COMMENT an offset applied to the V coordinate of the background texture
-	METHOD method_25435 getNarrationMessage ()Lnet/minecraft/class_2561;
+	METHOD method_25435 getTitleNarrationMessage ()Lnet/minecraft/class_2561;
```
Comment (liach) [inline mappings/net/minecraft/client/gui/screen/Screen.mapping]: Maybe `getNarratedTitle`?

File: mappings/net/minecraft/client/gui/screen/narration/ToStringTransformer.mapping [hunk 5]
```diff
@@ -0,0 +1,23 @@
+CLASS net/minecraft/class_6384 net/minecraft/client/gui/screen/narration/ToStringTransformer
+	FIELD field_33793 UNIT_TRANSFORMER Lnet/minecraft/class_6384;
```
Comment (liach) [inline mappings/net/minecraft/client/gui/screen/narration/ToStringTransformer.mapping]: This isn't a transformer, but data packed with transformation logic, so I would just name this to indicate this is a data class, like `Narration` (not idea either, as this is essentially supplying a stream of sentences to a string consumer)

File: mappings/net/minecraft/client/gui/screen/narration/Narration.mapping [hunk 6]
```diff
@@ -1,5 +1,5 @@
-CLASS net/minecraft/class_6384 net/minecraft/client/gui/screen/narration/ToStringTransformer
-	FIELD field_33793 UNIT_TRANSFORMER Lnet/minecraft/class_6384;
+CLASS net/minecraft/class_6384 net/minecraft/client/gui/screen/narration/Narration
+	FIELD field_33793 UNIT_NARRATION Lnet/minecraft/class_6384;
```
Comment (liach) [inline mappings/net/minecraft/client/gui/screen/narration/Narration.mapping]:
```suggestion
	FIELD field_33793 EMPTY Lnet/minecraft/class_6384;
```
This narration transforms data into no sentences.

File: mappings/net/minecraft/client/gui/screen/narration/Narration.mapping [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+	METHOD equals (Ljava/lang/Object;)Z
+		ARG 1 obj
+	METHOD method_37036 string (Ljava/lang/String;)Lnet/minecraft/class_6384;
+		ARG 0 string
+	METHOD method_37037 texts (Ljava/util/List;)Lnet/minecraft/class_6384;
+		ARG 0 texts
+	METHOD method_37038 (Ljava/util/List;Ljava/util/function/Consumer;Ljava/util/List;)V
+		ARG 1 consumer
+		ARG 2 texts
+	METHOD method_37039 transform (Ljava/util/function/Consumer;)V
```
Comment (liach) [inline mappings/net/minecraft/client/gui/screen/narration/Narration.mapping]:
```suggestion
	METHOD method_37039 forEachSentence (Ljava/util/function/Consumer;)V
```
this sounds more natural given this is now named `Narration`

**Review by liach — APPROVED** [review]

File: mappings/net/minecraft/client/gui/screen/Screen.mapping [hunk 8]
```diff
@@ -156,9 +156,7 @@ CLASS net/minecraft/class_437 net/minecraft/client/gui/screen/Screen
 		ARG 1 useTranslationsCache
 	METHOD method_37065 narrateScreen (Z)V
 		ARG 1 useTranslationsCache
-	METHOD method_37066 removeElement (Lnet/minecraft/class_364;)V
-		ARG 1 element
-	METHOD method_37067 clearChildren ()V
+	METHOD method_37067 removeAllChildren ()V
```
Comment (liach) [inline mappings/net/minecraft/client/gui/screen/Screen.mapping]: just use clearChildren, why this bad rename?

**Review by LambdAurora — APPROVED** [review]
Very much needed mappings.

**Comment by liach** [comment]
Hmm, can anyone verify the screen parts in this mapping? For the non-screen parts, this is quite correct.

File: mappings/net/minecraft/client/gui/screen/narration/Narration.mapping [hunk 9]
```diff
@@ -0,0 +1,23 @@
+CLASS net/minecraft/class_6384 net/minecraft/client/gui/screen/narration/Narration
+	FIELD field_33793 EMPTY Lnet/minecraft/class_6384;
+	FIELD field_33794 value Ljava/lang/Object;
+	FIELD field_33795 transformer Ljava/util/function/BiConsumer;
+	METHOD <init> (Ljava/lang/Object;Ljava/util/function/BiConsumer;)V
+		ARG 1 value
+		ARG 2 transformer
+	METHOD equals (Ljava/lang/Object;)Z
+		ARG 1 obj
```
Comment (enbrain) [inline mappings/net/minecraft/client/gui/screen/narration/Narration.mapping]:
```suggestion
		ARG 1 o
```
> "o" for the parameter of equals(Ljava/lang/Object;)Z methods

https://github.com/FabricMC/yarn/blob/3e70d627b71d8796c77877883a742907aed29fd7/CONVENTIONS.md#abbreviations

File: mappings/net/minecraft/client/gui/screen/Screen.mapping [hunk 10]
```diff
@@ -133,7 +136,40 @@ CLASS net/minecraft/class_437 net/minecraft/client/gui/screen/Screen
...
+	METHOD method_37056 addElementNarrations (Lnet/minecraft/class_6382;)V
+		ARG 1 builder
+	METHOD method_37058 setElementNarrationDelay (J)V
+		ARG 1 delayMs
+	METHOD method_37059 setScreenNarrationDelay (JZ)V
+		ARG 1 delayMs
+		ARG 3 restartElementNarration
+	METHOD method_37060 addDrawable (Lnet/minecraft/class_4068;)Lnet/minecraft/class_4068;
+		ARG 1 drawable
+	METHOD method_37061 getSelectedElementData (Ljava/util/List;Lnet/minecraft/class_6379;)Lnet/minecraft/class_437$class_6390;
```
Comment (enbrain) [inline mappings/net/minecraft/client/gui/screen/Screen.mapping]:
```suggestion
	METHOD method_37061 findSelectedElementData (Ljava/util/List;Lnet/minecraft/class_6379;)Lnet/minecraft/class_437$class_6390;
```

**Review by enbrain — APPROVED** [review]

**Pull request merged**